### PR TITLE
HPCC-16819 Add query publish and copy reporting of DFS not found issues

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by richardkchapman** [comment]
@wangkx please review

File: ecl/eclcmd/eclcmd_core.cpp [hunk 1]
```diff
@@ -463,7 +464,11 @@ class EclCmdPublish : public EclCmdWithEclTarget
         if (resp->getReloadFailed())
             fputs("\nAdded to Queryset, but request to reload queries on cluster failed\n", stderr);
 
-        return outputMultiExceptionsEx(resp->getExceptions());
+        int ret = outputMultiExceptionsEx(resp->getExceptions());
+        if (outputQueryFileCopyErrors(resp->getFileErrors()))
+            ret = 1;
```
Comment (wangkx) [inline ecl/eclcmd/eclcmd_core.cpp]: How about ret++ (adding 1 on the top of the exceptions)?

File: ecl/eclcmd/queries/ecl-queries.cpp [hunk 2]
```diff
@@ -540,6 +541,11 @@ class EclCmdQueriesCopy : public EclCmdCommon
 
         Owned<IClientWUQuerySetCopyQueryResponse> resp = client->WUQuerysetCopyQuery(req);
         int ret = outputMultiExceptionsEx(resp->getExceptions());
+        if (outputQueryFileCopyErrors(resp->getFileErrors()))
+            ret = 1;
```
Comment (wangkx) [inline ecl/eclcmd/queries/ecl-queries.cpp]: Similar to the last comment.

File: ecl/eclcmd/queries/ecl-queries.cpp [hunk 3]
```diff
@@ -540,6 +541,11 @@ class EclCmdQueriesCopy : public EclCmdCommon
...
+
+        return ret;
```
Comment (wangkx) [inline ecl/eclcmd/queries/ecl-queries.cpp]: why return here?

File: ecl/eclcmd/queries/ecl-queries.cpp [hunk 4]
```diff
@@ -693,9 +699,13 @@ class EclCmdQueriesCopyQueryset : public EclCmdCommon
         req->setAppendCluster(!optDontAppendCluster);
         req->setCopyFiles(!optDontCopyFiles);
         req->setAllowForeignFiles(optAllowForeign);
+        req->setIncludeFileErrors(true);
 
         Owned<IClientWUCopyQuerySetResponse> resp = client->WUCopyQuerySet(req);
         int ret = outputMultiExceptionsEx(resp->getExceptions());
+        if (outputQueryFileCopyErrors(resp->getFileErrors()))
+            ret = 1;
```
Comment (wangkx) [inline ecl/eclcmd/queries/ecl-queries.cpp]: ret++?

**Review by wangkx — COMMENTED** [review]
@afishbeck please see my comments.

File: ecl/eclcmd/eclcmd_core.cpp (hunk 1 above)
Comment (afishbeck) [inline ecl/eclcmd/eclcmd_core.cpp]: The return is an error code, not a counter.  1 means "general error".  If we wanted we could use very specific error codes, and warning codes.  0 is always OK, positive numbers are used as errors and negative numbers as warning.

File: ecl/eclcmd/queries/ecl-queries.cpp (hunk 3 above)
Comment (afishbeck) [inline ecl/eclcmd/queries/ecl-queries.cpp]: oops bad mege / cut past error, will fix.

**Comment by afishbeck** [comment]
@wangkx updated.  Please re-review.  I will squash after you've checked.

**Comment by wangkx** [comment]
@afishbeck looks good.

**Comment by richardkchapman** [comment]
@afishbeck Please squash

**Comment by afishbeck** [comment]
@richardkchapman squashed.

**Pull request merged**